plugins/wasmedge_stablediffusion/sd_func.cpp

```diff
@@ -396,28 +501,53 @@ Expect<uint32_t> SDImageToImage::body(
                     sample_method_t(SampleMethod), SampleSteps, Strength, Seed,
                     BatchCount, ControlImage, ControlStrength, StyleRatio,
                     NormalizeInput, InputIdImagesDir.data());
-  // TODO: upscale image
+  // Upscale image
+  if (UpscaleModelPathLen != 0) {
+    MEM_SPAN_CHECK(UpscaleModelSpan, MemInst, char, UpscaleModelPathPtr,
+                   UpscaleModelPathLen,
+                   "Failed when accessing the Upscaler Image memory."sv)
+    std::string UpscaleModelPath(UpscaleModelSpan.begin(),
+                                 UpscaleModelSpan.end());
+    upscalerModel(UpscaleModelPath.data(), UpscaleRepeats, NThreads, Wtype,
+                  BatchCount, Results);
+  }
+  // save results
   int Len;
   unsigned char *Png = stbi_write_png_to_mem(
-      reinterpret_cast<const unsigned char *>(Results->data), 0, Results->width,
+      reinterpret_cast<const unsigned char *>(Results), 0, Results->width,
       Results->height, Results->channel, &Len, nullptr);
-  if (OutputPathLen != 0) {
-    stbi_write_png(OutputPath.data(), Results->width, Results->height,
-                   Results->channel, Results->data, 0, nullptr);
+  size_t Last = OutputPath.find_last_of(".");
+  std::string DummyName = OutputPath;
+  if (Last != std::string::npos) {
+    DummyName = OutputPath.substr(0, Last);
+  }
+  for (uint32_t I = 0; I < BatchCount; I++) {
+    if (Results[I].data != nullptr) {
+      std::string FinalImagePath;
+      if (I <= 0)
+        FinalImagePath = DummyName + ".png";
```

Comment:
What if the OutputPath = `ABC.xyz` or `ABC.PNG`? You split the original file name into two parts: name and extension. However, you keep the name but replace the extension with `png` directly.